From 5e7233a1d27c7827b28cda4c9f55f911f518e59d Mon Sep 17 00:00:00 2001
From: Wolfgang Grandegger <wg@grandegger.com>
Date: Mon, 17 Feb 2020 21:53:11 +0100
Subject: [PATCH 3/5] ARM: dts: socfpga: Add support for the Aries MAX SoM on
 the MAXEVK board

Add support for Aries MAX Arria10 SoM, which is Arria10 based and the
associated Aries MAXEVK baseboard. The board can boot from eMMC or
the network. Ethernet, eMMC, UART and USB is supported.

Signed-off-by: Wolfgang Grandegger <wg@aries-embedded.de>
---
 arch/arm/boot/dts/Makefile                   |  1 +
 arch/arm/boot/dts/socfpga_arria10_max.dtsi   | 68 ++++++++++++++++++++
 arch/arm/boot/dts/socfpga_arria10_maxevk.dts | 41 ++++++++++++
 3 files changed, 110 insertions(+)
 create mode 100644 arch/arm/boot/dts/socfpga_arria10_max.dtsi
 create mode 100644 arch/arm/boot/dts/socfpga_arria10_maxevk.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 8337602bc930..488e7d541940 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1076,6 +1076,7 @@ dtb-$(CONFIG_ARCH_S5PV210) += \
 	s5pv210-torbreck.dtb
 dtb-$(CONFIG_ARCH_INTEL_SOCFPGA) += \
 	socfpga_arria5_socdk.dtb \
+	socfpga_arria10_maxevk.dtb \
 	socfpga_arria10_socdk_nand.dtb \
 	socfpga_arria10_socdk_qspi.dtb \
 	socfpga_arria10_socdk_sdmmc.dtb \
diff --git a/arch/arm/boot/dts/socfpga_arria10_max.dtsi b/arch/arm/boot/dts/socfpga_arria10_max.dtsi
new file mode 100644
index 000000000000..c775ca7f0ab7
--- /dev/null
+++ b/arch/arm/boot/dts/socfpga_arria10_max.dtsi
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2020 Wolfgang Grandegger <wg@aries-embedded.de>
+ */
+
+#include "socfpga_arria10.dtsi"
+
+/ {
+	model = "Aries MAX";
+	compatible = "altr,socfpga-arria10", "altr,socfpga";
+
+	chosen {
+		bootargs = "earlyprintk";
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory@0 {
+		name = "memory";
+		device_type = "memory";
+		reg = <0x0 0x80000000>; /* 2GB */
+	};
+
+	ref_033v: 033-v-ref {
+		compatible = "regulator-fixed";
+		regulator-name = "0.33V";
+		regulator-min-microvolt = <330000>;
+		regulator-max-microvolt = <330000>;
+	};
+
+	soc {
+		clkmgr@ffd04000 {
+			clocks {
+				osc1 {
+					clock-frequency = <25000000>;
+				};
+			};
+		};
+	};
+};
+
+&gpio1 {
+	status = "disabled";
+};
+
+&mmc {
+	status = "okay";
+	num-slots = <1>;
+	cap-sd-highspeed;
+	cap-mmc-highspeed;
+	broken-cd;
+	bus-width = <8>;
+};
+
+&eccmgr {
+	sdmmca-ecc@ff8c2c00 {
+		compatible = "altr,socfpga-sdmmc-ecc";
+		reg = <0xff8c2c00 0x400>;
+		altr,ecc-parent = <&mmc>;
+		interrupts = <15 IRQ_TYPE_LEVEL_HIGH>,
+			     <47 IRQ_TYPE_LEVEL_HIGH>,
+			     <16 IRQ_TYPE_LEVEL_HIGH>,
+			     <48 IRQ_TYPE_LEVEL_HIGH>;
+	};
+};
+
+&watchdog1 {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/socfpga_arria10_maxevk.dts b/arch/arm/boot/dts/socfpga_arria10_maxevk.dts
new file mode 100644
index 000000000000..e1b2be19f97a
--- /dev/null
+++ b/arch/arm/boot/dts/socfpga_arria10_maxevk.dts
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2020 Wolfgang Grandegger <wg@aries-embedded.de>
+ */
+
+/dts-v1/;
+#include "socfpga_arria10_max.dtsi"
+
+/ {
+	model = "Aries MAX EVK";
+	compatible = "aries,maxevk", "altr,socfpga-arria10", "altr,socfpga";
+
+	aliases {
+		ethernet0 = &gmac1;
+		ethernet1 = &gmac2;
+		serial0 = &uart0;
+	};
+};
+
+&gmac1 {
+	phy-mode = "rgmii";
+	phy-addr = <0>;
+	status = "okay";
+};
+
+&gmac2 {
+	phy-mode = "rgmii";
+	phy-addr = <0>;
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&usb1 {
+	status = "okay";
+	dr_mode = "host";
+	external-vbus-supply;
+	//disable-over-current;
+};
-- 
2.34.1

